arbnode/transaction_streamer.go

```diff
@@ -82,18 +85,21 @@ var DefaultTransactionStreamerConfig = TransactionStreamerConfig{
 	MaxBroadcasterQueueSize: 50_000,
 	MaxReorgResequenceDepth: 1024,
 	ExecuteMessageLoopDelay: time.Millisecond * 100,
+	TrackBlockMetadataFrom:  0,
 }
 
 var TestTransactionStreamerConfig = TransactionStreamerConfig{
 	MaxBroadcasterQueueSize: 10_000,
 	MaxReorgResequenceDepth: 128 * 1024,
 	ExecuteMessageLoopDelay: time.Millisecond,
+	TrackBlockMetadataFrom:  0,
 }
 
 func TransactionStreamerConfigAddOptions(prefix string, f *flag.FlagSet) {
 	f.Int(prefix+".max-broadcaster-queue-size", DefaultTransactionStreamerConfig.MaxBroadcasterQueueSize, "maximum cache of pending broadcaster messages")
 	f.Int64(prefix+".max-reorg-resequence-depth", DefaultTransactionStreamerConfig.MaxReorgResequenceDepth, "maximum number of messages to attempt to resequence on reorg (0 = never resequence, -1 = always resequence)")
 	f.Duration(prefix+".execute-message-loop-delay", DefaultTransactionStreamerConfig.ExecuteMessageLoopDelay, "delay when polling calls to execute messages")
+	f.Uint64(prefix+".track-block-metadata-from", DefaultTransactionStreamerConfig.TrackBlockMetadataFrom, "block number starting from which the missing of blockmetadata is being tracked in the local disk. Disabled by default")
```

Comment:
> the description is not clear to me:
> 
> * isn't `track-block-metadata-from` a block number from which we want to have block metadata stored in local db? I think that the tracking of missing block metadata is just an implementation detail.
> * can we mention that `(0 = disabled)`?

Yep its the blockNumber from when we can guarantee either we have the blockMetadata in our local DB or we have a tracker in place which can be used to fetch the missing ones. Cool I'll mention setting to 0 disables it.



> additionally, maybe we should make sure in `arbnode.Config.Validate()` that `BlockMetadataRebuilder` is enabled when `track-block-metadata-from != 0`, and error otherwise (or at least warn if that config combination (rebuilder disabled && `track-block-metadata-from` > 0) makes sense in some corner cases).

Initially I had this check but removed it because I thought node runners should have the freedom to choose when to start bulk syncing blockMetadata while always having a option to track missing ones. 
Does it makes sense? For example they might not have the bulkBlockMetadata APi in place, but still would like to track it. 

I lean towards having a warning here instead of erroring out, let me know.